### PR TITLE
ENH: expose additional kwargs in explode method to mirror GeoPandas API

### PR DESCRIPTION
**Pull request by ebkurtz**
Expose `index_parts`, `column`, and `ignore_index` parameters in the `.explode()` method to mirror GeoPandas API. This allows supressing FutureWarning for `index_parts`, see  #245

### CODE REVIEW

**Comment by TomAugspurger** [comment]
Can you define it on `Series` and `DataFrame` separately like in Dask?

- Series: https://github.com/dask/dask/blob/09dbeebbc1c8d5e40f7afc08258572ee897f36e9/dask/dataframe/core.py#L4050-L4053
- DataFrame: https://github.com/dask/dask/blob/09dbeebbc1c8d5e40f7afc08258572ee897f36e9/dask/dataframe/core.py#L5461-L5464

**Comment by TomAugspurger** [comment]
The CI failure looks unrelated. I'll take a look at that later.

**Comment by ebkurtz** [comment]
Thanks for the tip on defining them separately. I'll look at creating a unit test.

File: dask_geopandas/core.py [hunk 1]
```diff
@@ -553,6 +549,15 @@ class GeoSeries(_Frame, dd.core.Series):
 
     _partition_type = geopandas.GeoSeries
 
+    @derived_from(geopandas.geodataframe.GeoSeries)
```
Comment (jorisvandenbossche) [inline dask_geopandas/core.py:552]:
```suggestion
    @derived_from(geopandas.GeoSeries)
```

File: dask_geopandas/core.py [hunk 2]
```diff
@@ -807,6 +812,16 @@ def spatial_shuffle(
 
         return sorted_ddf
 
+    @derived_from(geopandas.geodataframe.GeoDataFrame)
```
Comment (jorisvandenbossche) [inline dask_geopandas/core.py:815]:
```suggestion
    @derived_from(geopandas.GeoDataFrame)
```

**Review by jorisvandenbossche — COMMENTED** [review]
Thanks for the PR! The code looks good, would indeed be good to add a basic test using one of the keywords.

**Comment by ebkurtz** [comment]
Would I add that as a new function in `test_core.py` are as additional code in the existing functions? I haven't done much with unit tests.

**Comment by TomAugspurger** [comment]
Yes, a new function. You might be able to adapt a geopandas test like https://github.com/geopandas/geopandas/blob/dc695ee93113af5d1a93a5740ecd792c2b9d19b3/geopandas/tests/test_geom_methods.py#L1003.

**Comment by ebkurtz** [comment]
Getting a lot of test failures after refreshing my fork. Looks like something with pyarrow strings?

**Comment by jorisvandenbossche** [comment]
> Getting a lot of test failures after refreshing my fork. Looks like something with pyarrow strings?

Apologies for the slow follow-up. Indeed those failures were cased by something else, which has been fixed in the meantime. 

I also added the equivalent change to the new dask-expr based implementation.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
> The unit tests for the `ignore_index` argument are failing.

Your fix to create the expected result manually with the different index was still failing for the older geopandas or dask version. But so just added a skip for those older versions, as I don't think it is worth complicating the test for that.

**Comment by jorisvandenbossche** [comment]
@ebkurtz Thanks again for the PR and apologies it took so long to get merged!

**Pull request merged**